scanner/decisions/0006-redhat-cvss-scores.md

```diff
@@ -0,0 +1,51 @@
+
+## Context
+
+[ClairCore](https://github.com/quay/claircore) provides us with CVSS scores from NVD in the form of a Vulnerability Report enrichment.
+These scores may be not accurate for Red Hat products, as Red Hat analyzes each CVE and, potentially,
+reassigns the CVSS score. For example: [CVE-2021-26291](https://access.redhat.com/security/cve/CVE-2021-26291).
+
+There is currently work being done in ClairCore to [adopt CSAF and VEX files for Red Hat vulnerability data](https://www.redhat.com/en/blog/vulnerability-exploitability-exchange-vex-beta-files-now-available),
+which will deprecate the [OVAL v2 feeds](https://access.redhat.com/security/data/oval/v2/). Though there are plans to
+expose Red Hat's CVSS scores once the adoption is complete, this work is not expected to be completed until after
+the StackRox 4.4 release is cut. So, we need a temporary solution.
+
+## Decision
+
+We want to minimize any divergences we have with upstream ClairCore, which means we do **not** want to fork
+the ClairCore repository. That being said, forking the repository is the simplest way to proceed here.
+
+We know there is already an effort to replace the current [Red Hat vulnerability updater](https://github.com/quay/claircore/blob/v1.5.20/rhel/updaterset.go)
+with an implementation which supports CSAF and VEX files, and we know Red Hat Product Security is definitely moving away from
+the OVAL v2 feeds in favor of CSAF and VEX. Therefore, we are assured this is a **temporary** fork of the ClairCore repository.
+
+This fork provides us with the simplest path forward. There are no plans to create a Red Hat CVSS score enricher in
+upstream ClairCore at this time, and creating our own out-of-tree enricher will be time-consuming and tricky, as
+CVSS scores are potentially also product-dependent instead of just CVE-specific.
+
+The simplest path forward is to utilize a field of the [VulnerabilityReport](https://github.com/quay/claircore/blob/v1.5.20/vulnerabilityreport.go)
+which Scanner v4/StackRox does not plan on utilizing at this time: `Severity` (not to be mistaken for `NormalizedSeverity`).
+This idea was conceived a few months back, but was [rejected from upstream ClairCore](https://github.com/quay/claircore/pull/919),
+as the team did not want to change the semantics of this field which may be used by consumers. For Scanner v4/StackRox,
+we know we are not using this field, so we do not mind overwriting it. As can be seen from the original pull request,
+implementing this is rather straightforward and quick to do.
+
+The [stackrox/claircore](https://github.com/stackrox/claircore) repository has been created for this purpose.
+We will recreate the origin pull request, merge it into this repository, and add a `replace` directive to this repository's
+`go.mod` to replace the `github.com/quay/claircore` repository with our fork.
+
+Again, we expect this to be temporary, as we know there is work in-progress to adopt CSAF and VEX files in favor of OVAL v2.
```

Comment:
We need a plan to drop the fork. In particular, what is the impact of the Severity field in the DB containing changes that the upstream RHEL updater won't be able to handle properly? If there is any risk, maybe forking only the updater to get a separate set of vulnerabilities in the DB, which we can discard instead.

Reply:
> In particular, what is the impact of the Severity field in the DB containing changes that the upstream RHEL updater won't be able to handle properly?

I do not believe there should be any impact, as ClairCore reads the latest versions of each vulnerability in the database, and I do not believe there is anything special about `Severity` (no particular parsing).

That being said, not maintaining an entire forked repo would be nice. See the updates I made to this ADR